### PR TITLE
[SofaSimulationCore]  Set default  TaskScheduler worker threads to idle state

### PR DESCRIPTION
**Pull request by fspadoni**
Set worker threads to idle state when there are no more tasks to execute.
It should fix the problem of the CPU busy when the simulation is  not running raised in PR #1862  

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: SofaKernel/modules/SofaSimulationCore/src/sofa/simulation/DefaultTaskScheduler.h [hunk 1]
```diff
@@ -235,7 +235,7 @@ namespace sofa  {
             
             unsigned m_workerThreadCount;
             
-            bool m_workerThreadsIdle;
+            volatile bool m_workerThreadsIdle;
```
Comment (alxbilger) [inline SofaKernel/modules/SofaSimulationCore/src/sofa/simulation/DefaultTaskScheduler.h:161]: Can you explain why `volatile` is desirable here?

Comment (fspadoni) [reply SofaKernel/modules/SofaSimulationCore/src/sofa/simulation/DefaultTaskScheduler.h:161]: If the task scheduler system works correctly there should be only one thread modifying this variable and multiple threads reading it. The volatile prevents the multiple threads to cache this variable.

**Comment by fredroy** [comment]
Linked to the PR #1862; can be merged IMO

**Comment by fredroy** [comment]
I cant solve the conflicts for you @fspadoni ... (no rights to push into your branch unfortunately)
The conflict comes from #2002 and you just need to add your change into `WorkerThread.cpp` (instead of `DefaultTaskScheduler.cpp`)

**Pull request merged**